### PR TITLE
fix: Follow model

### LINKED ISSUE #70: Getting Following Feeds
When I try to get the feeds that the current user is following I get null trying to access the data. If you look at the response from the request it shows there is indeed data but using `value[0].target ` or `value[0]` each returns null.
```
I/flutter (17428): *** Request ***
I/flutter (17428): uri: https://api.stream-io-api.com/api/v1.0/feed/timeline/Prhia3HUNgPesoCjkaEdEUPuOWZ2personal/following?limit=100&offset=0&api_key=hmwext26344q&location=unspecified
I/flutter (17428): method: GET
I/flutter (17428): responseType: ResponseType.json
I/flutter (17428): followRedirects: true
I/flutter (17428): connectTimeout: 10000
I/flutter (17428): sendTimeout: 0
I/flutter (17428): receiveTimeout: 10000
I/flutter (17428): receiveDataWhenStatusError: true
I/flutter (17428): extra: {}
I/flutter (17428): headers:
I/flutter (17428):  stream-auth-type: jwt
I/flutter (17428):  x-stream-client: stream-feed-dart-client-android-0.1.3
I/flutter (17428):  Authorization: eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyX2lkIjoiUHJoaWEzSFVOZ1Blc29DamthRWRFVVB1T1daMnBlcnNvbmFsIn0.L8Yu4YxwCb4JykadVwlwTJyidBhIerqGpWn1LAK0Sj4
I/flutter (17428): data:
I/flutter (17428): null
I/flutter (17428):
I/flutter (17428): *** Response ***
I/flutter (17428): uri: https://api.stream-io-api.com/api/v1.0/feed/timeline/Prhia3HUNgPesoCjkaEdEUPuOWZ2personal/following?limit=100&offset=0&api_key=hmwext26344q&location=unspecified
I/flutter (17428): statusCode: 200
I/flutter (17428): headers:
I/flutter (17428):  connection: keep-alive
I/flutter (17428):  cache-control: no-cache
I/flutter (17428):  x-ratelimit-reset: 1621443000
I/flutter (17428):  date: Wed, 19 May 2021 16:49:38 GMT
I/flutter (17428):  access-control-allow-origin: *
I/flutter (17428):  x-ratelimit-limit: 500
I/flutter (17428):  content-encoding: gzip
I/flutter (17428):  x-ratelimit-remaining: 499
I/flutter (17428):  content-type: application/json;charset=utf-8
I/flutter (17428):  access-control-max-age: 86400
I/flutter (17428):  server: nginx
I/flutter (17428):  access-control-allow-headers: x-requested-with, content-type, accept, origin, authorization, x-csrftoken, x-stream-client, stream-auth-type
I/flutter (17428):  access-control-allow-methods: GET, POST, PUT, PATCH, DELETE, OPTIONS
I/flutter (17428):  content-length: 192
I/flutter (17428): Response Text:
I/flutter (17428): {"results":[{"feed_id":"timeline:Prhia3HUNgPesoCjkaEdEUPuOWZ2personal","target_id":"user:WPpmnSSN1uURXNODPak2rlsCV332personal","created_at":"2021-05-14T19:58:27.274792063Z","updated_at":"2021-05-14T19:58:27.274792063Z"}],"duration":"1.62ms"}
I/flutter (17428):
I/flutter (17428): Following: Follow(null, null)
```

### PR DESCRIPTION
**Pull request by sachaarbonel**
fix #70
introduce Un/FollowRelation models for batch methods un/followMany and fix Follow model deserialization

### CODE REVIEW

File: packages/stream_feed/lib/src/core/models/follow.dart [hunk 1]
```diff
@@ -14,9 +14,11 @@ class Follow extends Equatable {
 
   /// The combination of feed slug and user id separated by a colon
   ///For example: flat:1
+  @JsonKey(name: 'feed_id')
```
Comment (xsahil03x) [inline packages/stream_feed/lib/src/core/models/follow.dart:17]: I think we should rather rename our fields back to `feed_id` and `target_id` as now we have a separate model for the relation

Comment (sachaarbonel) [reply packages/stream_feed/lib/src/core/models/follow.dart:17]: Good point! Also, we could parse `created_at` and `updated_at` maybe there are some use cases where you need to know the date you followed a feed

File: packages/stream_feed/lib/src/core/models/follow_relation.dart [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import 'package:equatable/equatable.dart';
+import 'package:json_annotation/json_annotation.dart';
+
+part 'follow_relation.g.dart';
+
+///
+@JsonSerializable()
+class FollowRelation extends Equatable {
+  ///
+  const FollowRelation(this.source, this.target);
```
Comment (xsahil03x) [inline packages/stream_feed/lib/src/core/models/follow_relation.dart:10]: I think requried named params will be better while creating object.

Comment (xsahil03x) [reply packages/stream_feed/lib/src/core/models/follow_relation.dart:10]:
```suggestion
  const FollowRelation({required this.source, required this.target});
```

**Comment by xsahil03x** [comment]
might need to update unfollow relation too

File: packages/stream_feed/lib/src/core/models/follow_relation.dart [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  /// Create a new instance from a json
+  factory FollowRelation.fromJson(Map<String, dynamic> json) =>
+      _$FollowRelationFromJson(json);
+
+  /// The combination of feed slug and user id separated by a colon
+  ///For example: flat:1
+  final String? source;
+
+  /// the id of the feed you want to follow
+  final String? target;
```
Comment (xsahil03x) [inline packages/stream_feed/lib/src/core/models/follow_relation.dart:21]: they both will never be null as we always need source and target in order to call batch methods.

Comment (sachaarbonel) [reply packages/stream_feed/lib/src/core/models/follow_relation.dart:21]: good catch

**Pull request merged**